src/account-info/AccountDetails.js

```diff
@@ -1,59 +1,148 @@
+    layoutStyle: ORIENTATION_PORTRAIT
+  };
+
+  mediumAvatarUrl = (avatarUrl) => {
+    const matches = new RegExp(/(\w+)\.png/g).exec(avatarUrl);
+
+    if (matches) {
+      return avatarUrl.replace(matches[0], `${matches[1]}-medium.png`);
+    } else {
+      return avatarUrl;
+    }
+  };
+
   handleChatPress = () => {
     const { email, doNarrow, popRoute } = this.props;
     doNarrow(privateNarrow(email));
     popRoute();
   };
 
-  render() {
-    const { avatarUrl, fullName, email, status, auth } = this.props;
+  handleOrientationChange = (event) => {
+    this.setState({
+      layoutStyle: this.props.orientation
+    });
+  };
+
+  renderAvatar = (width) => (
+    <Avatar
+      avatarUrl={this.mediumAvatarUrl(this.props.avatarUrl)}
+      name={this.props.fullName}
+      size={width}
+      status={this.props.status}
+      realm={this.props.auth.realm}
+      shape="square"
+    />
+  );
+
+  renderSendPMButton = () => (
+    <ZulipButton
+      style={styles.sendButton}
+      text="Send Private Message"
+      onPress={this.handleChatPress}
+      icon="md-mail"
+    />
+  );
+
+  renderUserDetails = () => (
+    <View style={styles.details}>
+      {this.props.status &&
+        <UserStatusIndicator status={this.props.status} style={styles.statusIndicator} />}
+      <Text style={styles.info}>{this.props.email}</Text>
+    </View>
+  );
+
+  renderLandscapeLayout = () => {
+    screenWidth = Dimensions.get('window').width;
 
     return (
-      <View style={styles.padding}>
-        <View style={styles.avatarWrapper}>
-          <Avatar
-            avatarUrl={avatarUrl}
-            name={fullName}
-            size={100}
-            status={status}
-            realm={auth.realm}
-          />
+      <View style={styles.containerLandscape} onLayout={this.handleOrientationChange}>
+        <View style={styles.avatarContainerLandscape}>
+          {this.renderAvatar(screenWidth / 2)}
+        </View>
+        <View style={[styles.infoContainerLandscape]}>
+          {this.renderUserDetails()}
+          {this.renderSendPMButton()}
         </View>
-        <View style={styles.details}>
-          <Text style={styles.info}>{fullName}</Text>
-          <Text style={styles.info}>{email}</Text>
+      </View>
+    );
+  };
+
+  renderPortraitLayout = () => {
+    screenWidth = Dimensions.get('window').width;
+
+    return (
+      <View style={styles.containerPortrait} onLayout={this.handleOrientationChange}>
+        {this.renderAvatar(screenWidth)}
+        <View style={styles.infoContainer}>
+          {this.renderUserDetails()}
+          {this.renderSendPMButton()}
         </View>
-        <ZulipButton
-          style={styles.logoutButton}
-          secondary
-          text="Chat"
-          onPress={this.handleChatPress}
-        />
       </View>
     );
+  };
+
+  render() {
+    const { layoutStyle } = this.state;
+
+    if (layoutStyle === ORIENTATION_LANDSCAPE) {
+      return this.renderLandscapeLayout();
```

Comment:
Extract these into two files called `something.landscape.js' and 'something.portrait.js'
Maybe they will be what exactly you have in `renderLandscapeLayout` and make them functional component.

And do extract a function with parameters (orientation, portraitComponent landscapeComponent)